libtrip-notification/src/main/java/com/mapbox/navigation/trip/notification/MapboxTripNotification.kt

```diff
@@ -1,29 +1,413 @@
+import com.mapbox.navigation.base.model.route.RouteConstants.IMPERIAL
+import com.mapbox.navigation.base.model.route.RouteConstants.STEP_MANEUVER_MODIFIER_LEFT
+import com.mapbox.navigation.base.model.route.RouteConstants.STEP_MANEUVER_MODIFIER_RIGHT
+import com.mapbox.navigation.base.model.route.RouteConstants.STEP_MANEUVER_MODIFIER_SHARP_LEFT
+import com.mapbox.navigation.base.model.route.RouteConstants.STEP_MANEUVER_MODIFIER_SHARP_RIGHT
+import com.mapbox.navigation.base.model.route.RouteConstants.STEP_MANEUVER_MODIFIER_SLIGHT_LEFT
+import com.mapbox.navigation.base.model.route.RouteConstants.STEP_MANEUVER_MODIFIER_SLIGHT_RIGHT
+import com.mapbox.navigation.base.model.route.RouteConstants.STEP_MANEUVER_MODIFIER_STRAIGHT
+import com.mapbox.navigation.base.model.route.RouteConstants.STEP_MANEUVER_MODIFIER_UTURN
+import com.mapbox.navigation.base.model.route.RouteConstants.STEP_MANEUVER_TYPE_ARRIVE
+import com.mapbox.navigation.base.model.route.RouteConstants.STEP_MANEUVER_TYPE_CONTINUE
+import com.mapbox.navigation.base.model.route.RouteConstants.STEP_MANEUVER_TYPE_DEPART
+import com.mapbox.navigation.base.model.route.RouteConstants.STEP_MANEUVER_TYPE_END_OF_ROAD
+import com.mapbox.navigation.base.model.route.RouteConstants.STEP_MANEUVER_TYPE_FORK
+import com.mapbox.navigation.base.model.route.RouteConstants.STEP_MANEUVER_TYPE_MERGE
+import com.mapbox.navigation.base.model.route.RouteConstants.STEP_MANEUVER_TYPE_NEW_NAME
+import com.mapbox.navigation.base.model.route.RouteConstants.STEP_MANEUVER_TYPE_NOTIFICATION
+import com.mapbox.navigation.base.model.route.RouteConstants.STEP_MANEUVER_TYPE_OFF_RAMP
+import com.mapbox.navigation.base.model.route.RouteConstants.STEP_MANEUVER_TYPE_ON_RAMP
+import com.mapbox.navigation.base.model.route.RouteConstants.STEP_MANEUVER_TYPE_ROTARY
+import com.mapbox.navigation.base.model.route.RouteConstants.STEP_MANEUVER_TYPE_ROUNDABOUT
+import com.mapbox.navigation.base.model.route.RouteConstants.STEP_MANEUVER_TYPE_ROUNDABOUT_TURN
+import com.mapbox.navigation.base.model.route.RouteConstants.STEP_MANEUVER_TYPE_TURN
+import com.mapbox.navigation.base.model.route.RouteProgress
+import com.mapbox.navigation.base.options.TripNavigationOptions
 import com.mapbox.navigation.base.trip.TripNotification
+import com.mapbox.navigation.utils.extensions.inferDeviceLanguage
+import com.mapbox.navigation.utils.extensions.inferDeviceLocale
+import com.mapbox.navigation.utils.formatter.distance.DistanceFormatter
+import com.mapbox.navigation.utils.formatter.distance.getUnitTypeForLocale
+import com.mapbox.navigation.utils.formatter.time.TimeFormatter.formatTime
+import java.util.Calendar
 
-class MapboxTripNotification : TripNotification {
-    override fun getNotification(): Notification {
-        TODO("not implemented")
+class MapboxTripNotification(
+    private val applicationContext: Context,
+    private val tripNavigationOptions: TripNavigationOptions,
+    private var currentRoute: Route
```

Comment:
Immutable `RouteOptions` could be passed instead. Should that be a part of the `TripNotification` interface contract as well?